### PR TITLE
Enhance azure data explorer scaler metric name with trigger.metadata.metricName

### PR DESCRIPTION
**Pull request by cdlliuy**
PR to address https://github.com/kedacore/keda/issues/3588

### CODE REVIEW

**Comment by cdlliuy** [comment]
add @v-shenoy  for review

File: pkg/scalers/azure_data_explorer_scaler.go [hunk 1]
```diff
@@ -118,7 +118,13 @@ func parseAzureDataExplorerMetadata(config *ScalerConfig, logger logr.Logger) (*
 	}
 
 	// Generate metricName.
-	metadata.MetricName = GenerateMetricNameWithIndex(config.ScalerIndex, kedautil.NormalizeString(fmt.Sprintf("%s-%s", adxName, metadata.DatabaseName)))
+	metadataName, _ := getParameterFromConfig(config, "metricName", false)
```
Comment (v-shenoy) [inline pkg/scalers/azure_data_explorer_scaler.go:121]:
```suggestion
	metricName, err := getParameterFromConfig(config, "metricName", false)
```

**Review by v-shenoy — CHANGES_REQUESTED** [review]

**Review by v-shenoy — COMMENTED** [review]
Update the CHANGELOG with this change.

Could you also add a unit test checking the metric name is being used?

**Comment by v-shenoy** [comment]
Hmm, there has been some discussion regarding the need for this parameter - https://github.com/kedacore/keda/pull/3182

@JorTurFer @zroubalik can give their thoughts about this.

**Comment by JorTurFer** [comment]
Let's discuss this in the issue, I wrote a comment there
https://github.com/kedacore/keda/issues/3588#issuecomment-1225547340

File: pkg/scalers/azure_data_explorer_scaler.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (cdlliuy) [reply pkg/scalers/azure_data_explorer_scaler.go:121]: @v-shenoy ,  updated in another approach. please help to review again.

**Comment by cdlliuy** [comment]
close this PR since the solution is changed

**Pull request closed without merging**